AssetConnector/asset_connector/base.py

```diff
@@ -0,0 +1,290 @@
+        return requests.Session()
+
+    @cached_property
+    def http_header(self) -> dict[str, str]:
+        """
+        Get the headers for the connector.
+
+        Returns:
+            dict: Headers
+        """
+        return {
+            "Authorization": f"Bearer {self.configuration.api_key}",
+            "Content-Type": "application/json",
+        }
+
+    def handle_api_error(self, error_code: int) -> str:
+        error = {
+            400: "Invalid request format",
+            401: "Unauthorized access",
+            404: "Connector not found",
+        }
+        return error.get(error_code, "An unknown error occurred")
+
+    def post_assets_to_api(self, assets: AssetList, asset_connector_api_url: str) -> dict[str, str] | None:
+        """
+        Post assets to the Sekoia.io asset connector API.
+        Args:
+            assets (AssetList): List of assets to post.
+            asset_connector_api_url (str): URL of the asset connector API.
+        Returns:
+            dict[str, str] | None: Response from the API or None if an error occurred.
+        """
+
+        try:
+            request_body = {
+                "assets": assets,
+            }
+
+            for attempt in self._retry():
+                with attempt:
+                    res: Response = self._http_session.post(
+                        asset_connector_api_url,
+                        json=request_body,
+                        timeout=30,
+                    )
+
+            if res.status_code != 200:
+                error_message = self.handle_api_error(res.status_code)
+                self.log(
+                    message=f"Error while pushing assets to Sekoia.io: {error_message}",
+                    level="error",
+                )
+                return None
+
+            self.log(
+                message=f"Successfully posted {len(assets)} assets to Sekoia.io asset connector API",
+                level="info",
+            )
+            return res.json()
+
+        except Exception as ex:
+            message = f"Failed to forward {len(assets)} assets to aseet connector api"
+            self.log(message=message, level="error")
+            self.log_exception(ex, message=message)
+
+    def push_assets_to_sekoia(self, assets: AssetList) -> None:
+        """
+        Push assets to the Sekoia.io asset connector API.
+        Args:
+            assets (AssetList): List of assets to push.
+        Returns:
+            None: If the assets were successfully pushed.
+        """
+
+        if not assets:
+            return []
+
+        asset_connector_uuid = self.connector_configuration_uuid
+        if sekoia_base_url := self.configuration.sekoia_base_url:
+            batch_api = urljoin(sekoia_base_url, "api/v1/asset-connectors")
+        else:
+            batch_api = urljoin(self.production_base_url, "api/v1/asset-connectors")
+
+        url = urljoin(batch_api, asset_connector_uuid)
```

Comment:
Can you move this part in a dedicated method and wrap the method with `@cached_property` as decorator? It will avoid recomputing the endpoint url every time the connector have to push assets.